docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+
+export default class Cat extends Component {
+  render() {
+    return (
+      <Text>Hello, I am your cat!</Text>
+    );
+  }
+}
+```
+
+You additionally import `Component` from React:
+
+```jsx
+import React, {Component} from 'react';
+```
+
+Your component starts as a class extending `Component` instead of as a function:
+
+```jsx
+class Cat extends Component {}
+```
+
+Class components have a `render()` function. Whatever is returned inside it is rendered as a React element:
+
+```jsx
+class Cat extends Component {
+  render() {
+    return <Text>Hello, I am your cat!</Text>;
+  }
+}
+```
+
+And as with function components, you can export your class component:
+
+```jsx
+export default function Cat() {
+  // element(s) go here!
+}
+```
+
+<block class="endBlock syntax" />
+
+> This is one of many ways to export your component. This kind of export works well with the Snack Player. However, depending on your app’s file structure, you might need to use a different convention. This [handy cheatsheet on JavaScript imports and exports](https://medium.com/dailyjs/javascript-module-cheatsheet-7bd474f1d829) can help.
+
+Now take a closer look at that `return` statement. `<Text>Hello, I am your cat!</Text>` is using a kind of JavaScript syntax that makes writing elements convenient: JSX.
+
+## JSX
+
+React and React Native use JSX, a syntax that lets you write elements inside JavaScript like so: `<Text>Hello, I am your cat!</Text>`. The React docs have [a comprehensive guide to JSX](https://reactjs.org/docs/introducing-jsx.html) you can reference to learn even more. Here’s how you would write the previous example with JSX:
+
+```SnackPlayer name=JSX
+import React from 'react';
+import { Text } from 'react-native';
+
+export default function Cat() {
+  return (
+    <Text>Hello, I am your cat!</Text>
+  );
+}
+```
+
+Because JSX is JavaScript, you can use variables inside it. Here you are declaring a name for the cat, `name`, and embedding it with curly braces inside `<Text>`.
+
+```SnackPlayer name=Curly%20Braces
+import React from 'react';
+import { Text } from 'react-native';
+
+export default function Cat() {
+  const name = "Maru";
+  return (
+    <Text>Hello, I am {name}!</Text>
+  );
+}
+```
+
+## Custom Components
+
+You’ve already met [React Native’s Core Components](intro-react-native-components). React lets you nest these components inside each other to create new components. These nestable, reusable components are at the heart of the React paradigm.
+
+For example, you can nest [`Text`](text) and [`TextInput`](textinput) inside a [`View`](view) below, and React Native will render them together:
+
+```SnackPlayer name=Custom%20Components
+import React from 'react';
+import { Text, TextInput, View } from 'react-native';
+
+export default function Cat() {
+  return (
+    <View>
+      <Text>Hello, I am...</Text>
+      <TextInput
+        style={{
+          height: 40,
+          borderColor: 'gray',
+          borderWidth: 1 }}
```

Comment:
Nit: `}}` on next line for style. In general, you can use formatting that https://prettier.io/ does.